plugins/modules/azure_rm_subnet.py

```diff
@@ -100,6 +100,23 @@
         choices:
             - Enabled
             - Disabled
+    delegations:
+        description:
+            - An array of delegations.
+        type: list
```

Comment:
This parameter should not be a list, but a dictionary, because a subnet can only have one dedicated service!

Reply:
@Fred-sun see this : https://docs.microsoft.com/en-us/cli/azure/network/vnet/subnet?view=azure-cli-latest#az_network_vnet_subnet_list_available_delegations
seems that azure cli support a list of delegations as a creation parameters

Reply:
Thanks @haiyuazhang , I see!

Reply:
@haiyuazhang : Thank you !
@Fred-sun : as per the reference given by @haiyuazhang , as response from azure subnet is list of delegations, but currently we can associate only one delegation to the subnet. So, I have added check for input delegations length here : 
https://github.com/ShwetaliBerad/azure/blob/issue-142/plugins/modules/azure_rm_subnet.py#L357
If you want input to module as dictionary , and process it as list while creation/updation and as return response, I will make changes. 
But , as currently azure returning delagations as list, in future they might support multiple delgation association to subnet. so I have made it as list.
Please do let me know if you want to change input type as dictionary.